sliceguard/modeling.py

```diff
@@ -107,3 +109,96 @@ def fit_classification_regression_model(
         raise RuntimeError("Could not run inference. Not valid task given.")
 
     return y_preds, y_probs, classes
+
+
+def _check_training_imports():
+    try:
+        from transformers import (
+            AutoImageProcessor,
+            AutoModelForImageClassification,
+            DefaultDataCollator,
+            TrainingArguments,
+            Trainer,
+        )
+        import torch
+    except ImportError:
+        raise Warning(
+            'Optional dependency required! (pip install "sliceguard[embedding]")'
+        )
+
+    return
+
+
+def _transform(example_batch, image_processor):
+    inputs = image_processor(
+        [x.convert("RGB") for x in example_batch["image"]], return_tensors="pt"
+    )
+    inputs["label"] = example_batch["label"]
+    return inputs
+
+
+def load_image_preprocessor(model_name):
+    """Load an image preprocessor for the model. This will resize the images to the correct size for the model."""
+    from transformers import AutoImageProcessor
+
+    checkpoint = model_name
+    image_processor = AutoImageProcessor.from_pretrained(checkpoint)
+    return image_processor
+
+
+def load_model(model_name, num_labels):
+    """Load the pre-trained model with AutoModelForImageClassification. Specify number of labels."""
+    from transformers import AutoModelForImageClassification
+
+    checkpoint = model_name
+    model = AutoModelForImageClassification.from_pretrained(
+        checkpoint, num_labels=num_labels
+    )
+    return model
+
+
+def finetune_image_classfier(
```

Comment:
Consider integrating this with fit_classification_regression_model. Right now this will always train a model on the generated embedding vectors. However, introducing an automl_model_name parameter that will call the training of the huggingface_image_classifier on the raw data would probably make sense.